core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java

```diff
@@ -93,7 +93,7 @@ else if ("Mac OS X".equals(osName)) {
 
     private static void verifyJavaVersion()
     {
-        Version required = Version.parse("17.0.3");
+        Version required = Version.parse("17.0.5");
```

Comment:
Why wasn't this also updated to require Java 21 ?
Since we've removed certain JVM configs (and related documentation) which were helpful on older versions like `-XX:+UseAESCTRIntrinsics` and `-XX:-G1UsePreventiveGC`, but aren't requiring Java 21, we now have the possibility of deployments which will use older JVM but miss these configs.
cc: @sopel39 @lukasz-stec

Reply:
This would force JDK 21 usage. If you want to use JDK 17 you need to use old configs

Reply:
> This would force JDK 21 usage

I know it would force JDK 21 usage, I'm asking why not ?

> If you want to use JDK 17 you need to use old configs
 
How does a user know which old configs should be used given we've scrubbed all mention of them from the repo ?
If we want to support both Java 17 and Java 21 simultaneously for some time, then our docs should clarify this and maintain the previous recommendations for Java 17 jvm configs.

Reply:
@raunaqmorarka existing other-than-docker deployments can still work without any changes if you stick to 17 and we can always revert (probability is low but still possible)

Reply:
I get that existing setups will continue to work, my concern is about new other-than-docker deployments, they would miss the old recommended JVM configs and would likely deploy with Java 17 as Java 21 is still very new.

Reply:
> Since we've removed certain JVM configs (and related documentation) which were helpful on older versions like -XX:+UseAESCTRIntrinsics and -XX:-G1UsePreventiveGC

Can we keep them even in 21 or they were completely removed in newest version of Java? I don't think they cause harm in 21